src/odemis/driver/xt_client.py

```diff
@@ -1092,12 +1094,17 @@ def __init__(self, name, role, parent, rng=None, **kwargs):
             rng["y"] = stage_info["range"]["y"]
         if "z" not in rng:
             rng["z"] = stage_info["range"]["z"]
+        if "t" not in rng:
+            rng["t"] = stage_info["range"]["t"]
+        if "r" not in rng:
+            rng["r"] = stage_info["range"]["r"]
 
         axes_def = {
-            # Ranges are from the documentation
-            "x": model.Axis(unit="m", range=rng["x"]),
-            "y": model.Axis(unit="m", range=rng["y"]),
-            "z": model.Axis(unit="m", range=rng["z"]),
+            "x": model.Axis(unit=stage_info["unit"]["x"], range=rng["x"]),
+            "y": model.Axis(unit=stage_info["unit"]["y"], range=rng["y"]),
+            "z": model.Axis(unit=stage_info["unit"]["z"], range=rng["z"]),
+            "t": model.Axis(unit=stage_info["unit"]["t"], range=rng["t"]),
+            "r": model.Axis(unit=stage_info["unit"]["r"], range=rng["r"]),
```

Comment:
Sorry, I realise only now: in Odemis the rotation axes are normally called rx (for "t") and "rz" (for "r"). So you need translate the axes names.

Reply:
done